ArcGISRuntimeSDKQt_CppSamples/Scenes/AddIntegratedMeshLayer/AddIntegratedMeshLayer.cpp

```diff
@@ -26,19 +26,38 @@
 #include "IntegratedMeshLayer.h"
 
 using namespace Esri::ArcGISRuntime;
+namespace
+{
+  const QUrl meshLyrUrl("https://tiles.arcgis.com/tiles/z2tnIkrLQ2BRzr6P/arcgis/rest/services/Girona_Spain/SceneServer");
+}
 
 AddIntegratedMeshLayer::AddIntegratedMeshLayer(QObject* parent /* = nullptr */):
   QObject(parent),
-  m_scene(new Scene(this))
+  m_scene(new Scene(this)),
+  m_integratedMeshLyr(new IntegratedMeshLayer(meshLyrUrl, this))
 {
-  // create the integrated mesh layer
-  const QUrl meshLyrUrl("https://tiles.arcgis.com/tiles/z2tnIkrLQ2BRzr6P/arcgis/rest/services/Girona_Spain/SceneServer");
-  IntegratedMeshLayer* integratedMeshLyr = new IntegratedMeshLayer(meshLyrUrl, this);
+  // Connect the doneLoading signal to the handleError() method.
+  connect(m_integratedMeshLyr, &IntegratedMeshLayer::doneLoading, this, &AddIntegratedMeshLayer::handleError);
+
+  m_scene->operationalLayers()->append(m_integratedMeshLyr);
 
-  // add the layer to the scene
-  m_scene->operationalLayers()->append(integratedMeshLyr);
+  setIntegratedMeshViewpoint();
+}
+
+void AddIntegratedMeshLayer::handleError(Error error)
+{
+  // If the doneLoading error message is not empty, an error has been thrown.
+  if (!error.isEmpty())
+  {
+    if (error.additionalMessage().isEmpty())
+      setErrorMessage(error.message());
+    else
+      setErrorMessage(error.message() + "\n" + error.additionalMessage());
```

Comment:
since this is the slot to handle error messages, it seems ideal to reasonably consolidate the logic into the slot and get rid of your setErrorMessage function.

Something like:

```suggestion

    if (error.additionalMessage().isEmpty())
      m_errorMessage = error.message();
    else
      m_errorMessage = error.message() + "\n" + error.additionalMessage();
    
    emit errorMessageChanged();
```